doc/releases/changelog-dev.md

```diff
@@ -209,6 +209,8 @@
 * Samples on lightning.qubit/kokkos can now be seeded with `qjit(seed=...)`.
   [(#1164)](https://github.com/PennyLaneAI/catalyst/pull/1164)
 
+* The compiler pass `-remove-chained-self-inverse` can now also cancel adjoints of arbitrary unitaries (on top of just the named Hermitian gates).
+  [(#1186)](https://github.com/PennyLaneAI/catalyst/pull/1186)
```

Comment:
There is a tricky usage of `qml.adjoint`, that it can act on an entire function (as opposed to just a single gate).

An idea @rmoyard and I discussed is to have an adjoint canonicalization pass before the peephole pipeline. The canonicalization would:
- remove the `adjoint` attribute on named Hermitian gates
- change to minus rotation angles for rotation gates
- change to Hermitian conjugate for unitary op
- apply the "$(U_1 U_2 U_3)^\dagger = U_3^\dagger U_2^\dagger U_1^\dagger$"

(Though, there is an argument to be made that this will make `cancel_inverse` harder)

Reply:
Note that the `--adjoint-lowering` already performs the $(U_1 U_2 U_3)^\dagger = U_3^\dagger U_2^\dagger U_1^\dagger$

https://github.com/PennyLaneAI/catalyst/blob/main/mlir/test/Quantum/AdjointTest.mlir

Reply:
Chatted some more. We think the following pipeline can be a default quantum peephole pipeline (to be put in the `--apply-transform-sequence`, aka if the user decides to turn on the default optimization pipeline this is the one executed, like -O2):

1. `--adjoint-lowering`, so that all the adjoints are on single gates and not on regions
2. `--cancel-inverse`, because it relies on abstract adjoint attribute
3. `--adjoint-canonicalization` to remove as many adjoint attributes as possible (doesn't exist yet, @rmoyard would like to work on this)
4. `--merge-rotation`, because now the "adjoint"-ed rotation gates have become regular rotation gates (with minus-signed angles)

This is also beneficial for the quantitative advantage epic as a whole since this can be a default pipeline for benchmarking (cc @josh146)